docs/rules/custom-error.md

```diff
@@ -0,0 +1,64 @@
+# Enforce correct Error subclassing
+
+Enforces the only valid way of `Error` subclassing.
+
+
+## Fail
+
+```js
+class CustomError extends Error {
+	constructor(message) {
+		super(message);
+		this.message = message;
+		this.name = 'CustomError';
+	}
+}
+```
+
+```js
+class CustomError extends Error {
+	constructor(message) {
+		super(message);
+	}
+}
+```
+
+```js
+class CustomError extends Error {
+	constructor(message) {
+		super();
+		this.message = message;
+		this.name = = 'CustomError';
+	}
+}
+```
+
+```js
+class CustomError extends Error {
+	constructor(message) {
+		super(message);
+		this.name = this.constructor.name;
+	}
+}
+```
```

Comment:
I think each of the examples should say why they're incorrect as it's probably not clear to the user.